roles/confluent.kafka_connect_replicator/tasks/main.yml

```
@@ -0,0 +1,300 @@
+    keystore_path: "{{kafka_connect_replicator_keystore_path}}"
+    keystore_storepass: "{{kafka_connect_replicator_keystore_storepass}}"
+    keystore_keypass: "{{kafka_connect_replicator_keystore_keypass}}"
+    service_name: confluent-replicator
+    user: "{{kafka_connect_replicator_user}}"
+    group: "{{kafka_connect_replicator_group}}"
+    hostnames: "{{ [inventory_hostname, ansible_fqdn, ansible_host, ansible_ssh_host] | unique }}"
+    ca_cert_path: "{{kafka_connect_replicator_ca_cert_path}}"
+    cert_path: "{{kafka_connect_replicator_cert_path}}"
+    key_path: "{{kafka_connect_replicator_key_path}}"
+    export_certs: "{{kafka_connect_replicator_export_certs}}"
+    ssl_ca_cert_file_path: "{{kafka_connect_replicator_ssl_ca_cert_path}}"
+    ssl_signed_cert_filepath: "{{kafka_connect_replicator_ssl_cert_path}}"
+    ssl_key_filepath: "{{kafka_connect_replicator_ssl_key_path}}"
+    ssl_key_password: "{{kafka_connect_replicator_ssl_key_password}}"
+    ssl_truststore_file_path: "{{kafka_connect_replicator_ssl_truststore_file_path}}"
+    ssl_keystore_file_path: "{{kafka_connect_replicator_ssl_keystore_file_path}}"
+  when: kafka_connect_replicator_ssl_enabled|bool or kafka_connect_replicator_ssl_provided_keystore_and_truststore|bool
+
+- name: Configure TLS for Kafka Connect Replicator Consumer
+  include_role:
+    name: confluent.ssl
+  vars:
+    truststore_storepass: "{{kafka_connect_replicator_consumer_truststore_storepass}}"
+    truststore_path: "{{kafka_connect_replicator_consumer_truststore_path}}"
+    keystore_path: "{{kafka_connect_replicator_consumer_keystore_path}}"
+    keystore_storepass: "{{kafka_connect_replicator_consumer_keystore_storepass}}"
+    keystore_keypass: "{{kafka_connect_replicator_consumer_keystore_keypass}}"
+    service_name: kafka-connect-replicator
+    user: "{{kafka_connect_replicator_user}}"
+    group: "{{kafka_connect_replicator_group}}"
+    hostnames: "{{ [inventory_hostname, ansible_fqdn, ansible_host, ansible_ssh_host] | unique }}"
+    ca_cert_path: "{{kafka_connect_replicator_consumer_ca_cert_path}}"
+    cert_path: "{{kafka_connect_replicator_consumer_cert_path}}"
+    key_path: "{{kafka_connect_replicator_consumer_key_path}}"
+    export_certs: "{{kafka_connect_replicator_consumer_export_certs}}"
+    ssl_ca_cert_file_path: "{{kafka_connect_replicator_consumer_ssl_ca_cert_path}}"
+    ssl_signed_cert_filepath: "{{kafka_connect_replicator_consumer_ssl_cert_path}}"
+    ssl_key_filepath: "{{kafka_connect_replicator_consumer_ssl_key_path}}"
+    ssl_key_password: "{{kafka_connect_replicator_consumer_ssl_key_password}}"
+    ssl_truststore_file_path: "{{kafka_connect_replicator_consumer_ssl_truststore_file_path}}"
+    ssl_keystore_file_path: "{{kafka_connect_replicator_consumer_ssl_keystore_file_path}}"
+  when: kafka_connect_replicator_consumer_ssl_enabled|bool or kafka_connect_replicator_consumer_ssl_provided_keystore_and_truststore|bool
```

```suggestion
  when: kafka_connect_replicator_consumer_ssl_enabled|bool
```

Is the or redundant here? I think `kafka_connect_replicator_consumer_ssl_enabled` must be true for `kafka_connect_replicator_consumer_ssl_provided_keystore_and_truststore` to be true.

---

Thats interesting about self signed certs not working, why is that?

---

By not working, we are not going to be generating them on the replicator deployment.  You may use self signed on your clusters, but then you still have to give us the certs and ca or keystore/truststore etc when deploying replicator.

---

One piece of functionality that I added w the Reconfigure work is - when using self signed certs if CA cert exists, by default it will not get updated on subsequent runs. 

I suspect w that feature, you could do self signed certs on all the kafka clusters and the replicator cluster and SSL will work bc all the clusters will have the same CA

Can we confirm self signed certs do not work after merging in my replicator work?

---

I don't think we are understanding each other here.  When you stand up a cluster, we offer 3 options, we can generate self signed certs for you, you can provide us with certs, or you can provide us with keystores/truststores.

Out of those 3 options the 1st option does not apply to replicator, as you are connecting to already existing clusters.  In which case, there is no need to have replicator generate self signed certs, you need to provide us the certs that match with the CA on your cluster or keystore/truststore which contain them.

---

We have multiple use cases when we have existing kafka clusters provisioned with "self signed certs", where external clusters also work with "self signed certs"- centralized audit logs and centralized rbac for example. why is replicator unique?

---

You are right we do, and those certs are not generated on the client side, they are generated on the cluster.  The relevant certs are then passed on to the clients to be used.

In other words, you aren't going to use Replicator to generate CA's and Certs, and then update your cluster to use them, you are going to do it the other way around.  Thus there is no need for Replicator role to have the ability to generate self signed certs.

---

Yes but if you have cp-ansible generate a CA and certs for cluster 1. and then cp-ansible use that same CA to generate certs for cluster 2 and then have cp-ansible use that same CA to generate certs for the replicator cluster then that follows your pattern. And that is how the self signed certs functionality i added in 61x branch works

---

Right.  But you need to give replicator that CA to work with, which is what I am saying here.  We don't want replicator to have the ability to generate a new CA and sign certs with that CA.  Thus that leaves you with two out of the three options for SSL that we allow on cluster deployment.

---

You will not need to give replicator the CA. It gets generated when provisioning cluster 1 and will reside in the generated_ssl dir. by the time replicator provisioning starts, cp-ansible will see the CA in the generated_ssl dir and decide not to regenerate it.

Code here: https://github.com/confluentinc/cp-ansible/blob/6.1.x/tasks/certificate_authority.yml

---

Sure, I understand that.  However this assumes you are deploying Replicator at the same time/with the same access as one/both clusters.  That is not the typical Replicator deployment model.  The typical deployment model is that you already have at least one cluster, and more often than not, the second as up as well.  Then, you deploy Replicator.  Generally Replicator will also be locked down as will the clusters so they wont have the ability to share access to that generation dir.  Clusters are typically in completely separate DCs.

---

Great point. I do not see why we should omit a functional use case though. Typically the self signed certs option is for demos/ making quick provisioning easy. So I would say that in the demo case users would likely do the provisioning at the same time. Essentially we are unnecessarily making replicator demo environments more challenging than they could be.

When we start discussing locked down clusters in multiple data centers then i doubt those users will even be using self signed certs anyhow